enterprise/server/backends/authdb/authdb.go

```diff
@@ -748,6 +789,8 @@ func (d *AuthDB) GetAPIKeyForInternalUseOnly(ctx context.Context, groupID string
 		SELECT * FROM "APIKeys"
 		WHERE group_id = ?
 		AND (user_id IS NULL OR user_id = '')
+		AND impersonation = 0
```

Comment:
Would there be any benefit to having this predicate be `WHERE impersonation = <is_impersonating>` instead?

Reply:
No, I don't see a case at this time where we'd want to (internally) pick a impersonation key over a different group key.